libs/hwes/eventstoredb/aggregate_store.go

```diff
@@ -0,0 +1,158 @@
+package eventstoredb
+
+import (
+	"context"
+	"errors"
+	"github.com/EventStore/EventStore-Client-Go/esdb"
+	"hwes"
+	"hwutil"
+	"io"
+	"math"
+)
+
+type AggregateStore struct {
+	es *esdb.Client
+}
+
+type getExpectedRevision = func(ctx context.Context, aggregate hwes.Aggregate) (esdb.ExpectedRevision, error)
+
+func NewAggregateStore(es *esdb.Client) *AggregateStore {
+	return &AggregateStore{es: es}
+}
+
+func (a *AggregateStore) Load(ctx context.Context, aggregate hwes.Aggregate) error {
+	stream, err := a.es.ReadStream(ctx, aggregate.GetStreamID(), esdb.ReadStreamOptions{}, math.MaxUint64) // MaxUint64 for "all" events
+	if err != nil {
+		return err
+	}
+	defer stream.Close()
```

Comment:
move this in front of the error handling for good measure

Reply:
Oh yes! Thats quite important!